src/legacyEdge/edgeDebugAdapter.ts

```diff
@@ -364,22 +364,23 @@ export class EdgeDebugAdapter extends CoreDebugAdapter {
         this._scriptParsedEventBookKeeping = {};
     }
 
-    protected async sendLoadedSourceEvent(script: Crdp.Debugger.ScriptParsedEvent): Promise<void> {
+    protected async sendLoadedSourceEvent(script: Crdp.Debugger.ScriptParsedEvent, reason: LoadedSourceEventReason = null): Promise<void> {
         // If navigation is in progress, we wait for it to complete before sending any new script loaded events
         // This is done because in case of quick refreshes, we end up sending 'changed' events for new scripts because
         // scriptParsedEventBookKeeping hasn't been refreshed yet
         if (!this._navigationInProgress) {
-            let loadedSourceReason: LoadedSourceEventReason;
-            if (!this._scriptParsedEventBookKeeping[script.scriptId]) {
-                this._scriptParsedEventBookKeeping[script.scriptId] = true;
-                loadedSourceReason = 'new';
-            } else {
-                loadedSourceReason = 'changed';
+            if (!reason) {
```

Comment:
What is the reason we receive from Edge if its not 'new' or 'changed'?

Reply:
We don't get any reason from Edge directly. We get one or more `ScriptParsedEvent` events from Edge, and if we haven't seen a parse event for that script yet, we send the `LoadedSourceEvent` to VS/VS code with reason "new", and if we have seen it before, we send it with reason "changed".

The other possible reason is "removed", which we send after we get the `Runtime.executionContextsCleared` event from Edge.